### PR TITLE
[WIP] [DO NOT MERGE] px4fmu-v2/v3 sensor issues

### PR DESCRIPTION
**Pull request by dagar**
Powering off the sensors before reboot helps on most boards and avoids the bootloop, but after reboot not all sensors are coming back on the holybro (lsm303, l3gd20).

https://github.com/PX4/Firmware/pull/8703
https://github.com/PX4/Firmware/issues/8643
https://github.com/PX4/Firmware/issues/8529

 - power cycle sensors before commanded  reboot
 - fixes missing lsm303d on px4fmu-v3 (configs sorted to prevent future accidental exclusion like this)

### CODE REVIEW

File: src/drivers/boards/px4fmu-v2/px4fmu_init.c [hunk 1]
```diff
@@ -185,6 +185,13 @@ __EXPORT void board_on_reset(int status)
...
 
 	if (status >= 0) {
+		/* on reboot (status >= 0) power off sensors and peripherals */
+		stm32_configgpio(GPIO_VDD_3V3_SENSORS_EN);
+		stm32_gpiowrite(GPIO_VDD_3V3_SENSORS_EN, 0);
+
+		stm32_configgpio(GPIO_VDD_5V_PERIPH_DIS);
+		stm32_gpiowrite(GPIO_VDD_5V_PERIPH_DIS, 1);
+
 		up_mdelay(400);
```
Comment (LorenzMeier) [inline src/drivers/boards/px4fmu-v2/px4fmu_init.c]: Increasing this might unstuck the sensors.

Comment (dagar) [reply src/drivers/boards/px4fmu-v2/px4fmu_init.c]: No luck so far.

**Comment by dagar** [comment]
I need to do more testing, but I believe the only remaining issue applies to the defective boards with the R635/R603 problem (https://github.com/PX4/Firmware/issues/8643#issuecomment-358006829) not working correctly after a reboot command. Not great, but certainly better than getting stuck in a bootloop.

If we can figure out a workaround that puts these sensors into a good state for reboot or clean initialization without first resetting the 3.3V rail we can be done with this and move on.

**Comment by LorenzMeier** [comment]
That sounds like we would want this in.

**Comment by svpcom** [comment]
What will happens on unclean reset after nuttx hardfault?

**Comment by svpcom** [comment]
I've found a solution that works for me:

**Comment by dagar** [comment]
Works for you for which problem on which board?

**Comment by svpcom** [comment]
This fix the problem of half sensors init after reboot on mRo pixhawk and aliexpress pixhawk v2.4.6.

**Comment by davids5** [comment]
Do we have schematics for the board? The change suggest that the unused GPIO is pulled up by an external resistor but this may be an artifact of the last config on the pin before reset. The latter would imply we can fix it with a reset to a known state prior to detection.

**Comment by svpcom** [comment]
@davids5 But reset can be from hardfault without any GPIO cleanup before.

**Comment by davids5** [comment]
@svpcom Are you able to reproduce the error with a `hardfault_log fault 1` command and not with a reset button push or power cycle?

**Comment by svpcom** [comment]
Yes, it has broken ``HW version: 0x0009000B`` (and half sensors available) after hardfault. I've tried on aliexpress pixhawk v2.4.6 using pr-reboot_sensors branch. After ``reboot`` command HW version is good:  ``HW version: 0x00090008``

**Comment by dagar** [comment]
Continued in https://github.com/PX4/Firmware/pull/9031

**Pull request closed without merging**